File: android/app/src/main/java/my/ssdid/wallet/domain/transport/dto/RegistryDtos.kt
package my.ssdid.wallet.domain.transport.dto

import kotlinx.serialization.Serializable
import my.ssdid.wallet.domain.model.DidDocument
import my.ssdid.wallet.domain.model.Proof

@Serializable
data class RegisterDidRequest(
    val did_document: DidDocument,
    val proof: Proof
)

@Serializable
data class RegisterDidResponse(
    val did: String,
    val status: String
)

@Serializable
data class UpdateDidRequest(
    val did_document: DidDocument,
    val proof: Proof
)

@Serializable
data class DeactivateDidRequest(
    val proof: Proof
)

@Serializable
data class ChallengeResponse(
    val challenge: String,
    val expires_at: String? = null,
    val domain: String? = null
)

@Serializable
data class RegistryInfoResponse(
    val name: String,
    val version: String,
    val did_method: String,
    val supported_algorithms: List<String> = emptyList(),
    val supported_proof_types: List<String> = emptyList(),
    val policies: RegistryPolicies? = null
)

@Serializable
data class RegistryPolicies(
    val proof_max_age_seconds: Int? = null
)

File: ios/SsdidWallet/Domain/Transport/DTO/RegistryDtos.swift
import Foundation

/// Request/response DTOs for Registry API interactions.

struct RegisterDidRequest: Codable {
    let didDocument: DidDocument
    let proof: Proof

    enum CodingKeys: String, CodingKey {
        case didDocument = "did_document"
        case proof
    }
}

struct RegisterDidResponse: Codable {
    let did: String
    let status: String
}

struct UpdateDidRequest: Codable {
    let didDocument: DidDocument
    let proof: Proof

    enum CodingKeys: String, CodingKey {
        case didDocument = "did_document"
        case proof
    }
}

struct DeactivateDidRequest: Codable {
    let proof: Proof
}

struct ChallengeResponse: Codable {
    let challenge: String
    var expiresAt: String? = nil
    var domain: String? = nil

    enum CodingKeys: String, CodingKey {
        case challenge
        case expiresAt = "expires_at"
        case domain
    }
}

struct RegistryInfoResponse: Codable {
    let name: String
    let version: String
    let didMethod: String
    var supportedAlgorithms: [String] = []
    var supportedProofTypes: [String] = []
    var policies: RegistryPolicies? = nil

    enum CodingKeys: String, CodingKey {
        case name, version
        case didMethod = "did_method"
        case supportedAlgorithms = "supported_algorithms"
        case supportedProofTypes = "supported_proof_types"
        case policies
    }
}

struct RegistryPolicies: Codable {
    var proofMaxAgeSeconds: Int? = nil

    enum CodingKeys: String, CodingKey {
        case proofMaxAgeSeconds = "proof_max_age_seconds"
    }
}
